contrib/xor-blocks/src/main.rs

```diff
@@ -0,0 +1,130 @@
+use std::{
+    env, fs,
+    io::{self, Read},
+    path::Path,
+    time::{Duration, Instant},
+};
+
+const MAGIC: [u8; 4] = [0xf9, 0xbe, 0xb4, 0xd9];
+
+fn main() -> Result<(), io::Error> {
+    let start = Instant::now();
+
+    let blocks_path = if env::args().len() > 1 {
+        env::args().nth(1).expect("the arg to exist").into()
+    } else {
+        #[allow(deprecated)]
```

Comment:
Have you tested that this picks the right one on Windows? That may be a good reason for the attribute.

Reply:
I have unfortunately not tested on Windows. If anyone could test I would be grateful. If this is the wrong home dir though then a user can just specify it manually using the `-datadir=` option.

It seems this function will be undeprecated in a future release https://github.com/rust-lang/rust/issues/132650.

Reply:
@1440000bytes thanks for testing! Unfortunately this tool will only work for mainnet. It looks like I also have to rename `Local` to `Roaming` in the Windows default datadir.

Reply:
@1440000bytes do you know which version switched to `Local`? If it was 28 or 29 then we only have to support `Roaming`.